crates/bevy_math/src/sampling/shape_sampling.rs

```diff
@@ -215,6 +215,60 @@ impl ShapeSample for Triangle3d {
     }
 }
 
+impl ShapeSample for Tetrahedron {
+    type Output = Vec3;
+
+    fn sample_interior<R: Rng + ?Sized>(&self, rng: &mut R) -> Self::Output {
+        let [v0, v1, v2, v3] = self.vertices;
+
+        // Generate a random point in a cube:
+        let mut coords: [f32; 3] = [
+            rng.gen_range(0.0..1.0),
+            rng.gen_range(0.0..1.0),
+            rng.gen_range(0.0..1.0),
+        ];
+
+        // The cube is broken into six tetrahedra of the form 0 <= c_0 <= c_1 <= c_2 <= 1,
+        // where c_i are the three euclidean coordinates in some permutation. (Since 3! = 6,
+        // there are six of them). Sorting the coordinates folds these six tetrahedra into the
+        // tetrahedron 0 <= x <= y <= z <= 1 (i.e. a fundamental domain of the permutation action).
+        coords.sort_by(|x, y| x.partial_cmp(y).unwrap());
+
+        // Now, convert a point from the fundamental tetrahedron into barycentric coordinates by
+        // taking the four succesive differences of coordinates; note that these telescope to sum
+        // to 1, and this transformation is linear, hence has constant Jacobian, hence preserves
+        // relative probability density.
+        let (a, b, c, d) = (
+            coords[0],
+            coords[1] - coords[0],
+            coords[2] - coords[1],
+            1. - coords[2],
+        );
+
+        v0 * a + v1 * b + v2 * c + v3 * d
+    }
+
+    fn sample_boundary<R: Rng + ?Sized>(&self, rng: &mut R) -> Self::Output {
+        // Filter out degenerate faces, since there are many ways a tetrahedron can degenerate.
+        let triangles: Vec<Triangle3d> = self
+            .faces()
+            .into_iter()
+            .filter(|t| !t.is_degenerate())
+            .collect();
+
+        if !triangles.is_empty() {
+            // Nondegenerate triangles have nonzero area, so this unwrap never fails.
+            let dist = WeightedIndex::new(triangles.iter().map(|t| t.area())).unwrap();
+            let idx = dist.sample(rng);
+            triangles[idx].sample_interior(rng)
+        } else {
+            // In this branch the triangle has zero surface area; just return a point that's on
```

Comment:
You mean the tetrahedron (therefore all triangles) have zero surface area.
```suggestion
            // In this branch the tetrahedron has zero surface area; just return a point that's on
```

```suggestion
            // In this branch all the triangles have zero surface area; just return a point that's on
```